### DISCUSSION TITLE
BitOps Community Meeting (09 Nov 2022): PR Best Practices, v2.2.0 release, Features & Development

### DISCUSSION THREAD

**Discussion by arm4b** [bitovi/bitops#351] [Announcements]
We invite everyone (developers, adopters, or just curious people) to join the bi-weekly BitOps Community `1 hour` Meeting on Wednesday, `09 Nov 2022, 12:00 PM US East` to discuss the ongoing project developments, updates, and plans, share use cases and stories or ask/answer questions about BitOps tool.

See https://github.com/bitovi/bitops/discussions/288 for more info on how to join.

### Follow-ups from the previous #331 Meeting
- BitOps landing page (@mickmcgrath13 @armab)
  - BitOps 1-3min intro Video (@PhillypHenning)
- Update [BitOps HeyEmoji blog](https://www.bitovi.com/blog/heyemoji-slack-app-aws-terraform-ansible) to v2 (@armab)
- Update [Combine Terraform and Ansible to Provision and Configure a Web Server](https://www.bitovi.com/blog/bitops-terraform-ansible) to v2 (@LeoDiazL)
- Trying Twitch: BitOps HeyEmoji demo

### Discussion: PR Reviews and Best practices
Pull Requests are reflection of the engineering culture. They are the core communication mechanism in the engineering team. Share your tips on Pull Requests, Reviews, and best practices, examples, and where we want to go together on that.

### Experiment: BitOps CLI - Ops Repo generator
@PhillypHenning did a quick experiment writing a BitOps Golang CLI tool: https://github.com/bitovi/bitops/pull/348 @PhillypHenning to share his Golang learnings and give the team an overview.
Per https://github.com/bitovi/bitops/issues/173 discuss the Python vs Golang path to BitOps CLI: pros & cons.

### v2.2.0 Release Preparation
Planning & coordination for the upcoming v2.2.0 release. Project: https://github.com/bitovi/bitops/milestone/8
What do we want to include in this release and when do we want to cut it.

### Feature: Secrets Masking
Initial take by @mickmcgrath13 https://github.com/bitovi/bitops/pull/345 and the discussion in https://github.com/bitovi/bitops/issues/208 about the secrets masking feature about different approaches, pros/cons and what do we want in this iteration.

### Development: Local plugin install (@mickmcgrath13)
@mickmcgrath13 to highlight https://github.com/bitovi/bitops/pull/349 new feature which allows setting local `file://` for plugin install instead of remote git clone.

### Development: Random Issue
If time allows, pick up one of the features from the [Development Roadmap](https://github.com/bitovi/bitops/discussions/245) and discuss it: design, implementation, UX, ideas, technical details.

**Comment by mickmcgrath13** [comment]
# 2022-11-09 Call Notes

## Follow-ups
- BitOps videos (5 parts. need to be spliced together)
  - good for youtube and docs. maybe need something a little more compact for the landing page
- HeyEmoji blog
- Combine TF and Ansible blog (update content so it works)
  - @LeoDiazL  working on the ops repo implementation, first
  - use the CLI tool?
  - Maybe do it manually first, and then have a section at the bottom
- Twitch (heyemoji)
  - mick to run the twitch (create a plugin with bitops?)

## PR review culture/best practices
- Draft PRs
- Understanding who you can work with on changes
- Local linting
- Scope your PRs
- specifically list everything you've done (even if there's additional small things)
  - main thing/reason for the PR linked to an issue
  - also section
- Squash & Merge vs rebase & cleanup locally before merge
  - rebase & cleanup is more overhead, but maybe a better approach. also relies on well-written commits
- Clean commits (commits should be specific to the change in the commit)
- Clear PR templates
  - they can be heavy. it should be clear how to remove which details aren't relevant
- Who can approve?
  - ideally, someone else approves
  - avoid self review if possible
- test the code (in the PR, clearly outline how to test)

@armab: Let's learn, share experience, and have fun together!


## Other Blogs
- cli tool blog
- deployment repo structure blog

## CLI tool
- Golang
  - fun language
  - interesting variable structure
  - easy to pick up
  - not a lot of support online (you'll be thinking through problems a lot yourself)
  - need help breaking up functions
- benefits to golang
  - portability (compatible with many OSs without any extra work)
  - might pique the interest of the community (vs python or something else)
- cons to golang
  - another language to maintain
- Python
  - need some way to write once and run anywhere
- docker image?
  - cons
    - user experience is challenging (have to write a long command)
  - pros
    - user experience is maybe actually okay because it's a similar way to interact as the way the deployment functionality
- what about having bitops run auto create things for you?

## 2.2 release
planned for next week

## Misc
- [Useful bitops commands](https://bitovi.slack.com/archives/C0129U3GL93/p1667586386774339)

## Next Time
- BitOps Udemy course
- Feature: Secrets Masking
- Development: Local plugin install
